File: STT3850/notes/2-13-25.py
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
np.random.seed(42)
# Generate random samples from exponential distribution
scale = 20 # Scale parameter (mean = 1/scale)
size = 1000000 # Number of samples
population = np.random.exponential(scale, size)
# Set the number of samples and sample size
num_samples = 100
sample_size = 50
# Generate sample means
sample_means = []
for _ in range(num_samples):
    sample = np.random.choice(population, sample_size,
    replace=False)
    sample_means.append(np.mean(sample))
# Plot the distribution of the population and the sample means
fig, axes = plt.subplots(1, 1, figsize=(8, 4))
sns.histplot(sample_means, ax=axes, kde=True)
axes.set_title('Distribution of Sample Means')
axes.set_xlabel('Sample Mean')
axes.set_ylabel('Frequency')
plt.show()

File: STT3850/notes/2-13-25.R
#! /usr/bin/R
set.seed(25)
times <- rexp(1000,1/30)

head(times)
mean(times)

times <- rexp(1000,1/15)
head(times)


times <- rexp(1000,1/6)
head(times)

waiting_time <- rexp(1000,1/30)

# If you get an exponential distribution, plotting it gives an exponential graph.
# If you plot the means, you get an approximately normal graph (Central Limit Theorem)
# x ~ N(1/l, (1/l)/n) -> n is the sample size, l -> lambda

rate_parameter <- 1/20

sample_size <- 30
sample_count <- 1000

sample_means <- rep(NA, sample_count)

# Generate samples and calculate means

set.seed(25)

for (i in 1:sample_count) {
	sample <- rexp(50,rate_parameter) # Generate an exponential sample
	sample_means[i] <- mean(sample)# Calculate sample mean
}

hist(sample_means, breaks = 20, main = "Distribution of Sample Means", xlab = "Sample Mean")
